app/controllers/admin/student_accounts_controller.rb

```diff
@@ -0,0 +1,37 @@
+require 'date'
+
+module Admin
+    class StudentAccountsController < BaseController
+      layout 'admin'
+  
+      def actions
+      end
+  
+      def get_number_of_accounts
+        days_num = get_number_of_days
+        accounts_num = query(days_num)
+        flash[:notice] = "The number of student acccounts created since July 1 is " + accounts_num.to_s
+        redirect_to actions_admin_student_accounts_path
+      end
+
+      def get_number_of_days
+        # number of days since July 1
+        today = Date.new
+        year = today.year
+        month = today.month
+        if month < 6
+            year = year - 1
+        end
+        #July is a keyword, so using month7
+        month7 = Date.parse year.inspect + "-07-01"
+        return (month7...today).count
+      end
+
+      def query(days_num)
+        role_type = :student
+        user_ids = User.where(role: role_type)
+        return SecurityLog.where(event_type: :sign_up_successful, user_id: user_ids).where('created_at > ?', days_num.days.ago).count
```

Comment:
Actually, `ActiveRecord` can take a `Date` object and automagically insert it properly into a query, so you don't need to calculate the number of days ahead of time. You can just do `SecurityLog.where('created_at >= ?', Date.parse('2019-07-01')).count`. And therefore, you can get rid of the `get_number_of_days` function.

Another thing, I would use `created_at >= ?` July 1st instead of `created_at > ?` July 1st to include those accounts created that day.

Reply:
Requested change has been made